external/fv3fit/fv3fit/pytorch/graph/mpg_unet.py

```diff
@@ -0,0 +1,331 @@
+        pooling_stride: pooling layer stride
+        activation: activation function
+    """
+
+    num_step_message_passing: int
+    edge_hidden_features: int
+    depth: int = 1
+    min_filters: int = 4
+    aggregator: str = "mean"
+    pooling_size: int = 2
+    pooling_stride: int = 2
+    activation: ActivationConfig = dataclasses.field(
+        default_factory=lambda: ActivationConfig()
+    )
+
+    def build(
+        self, in_channels: int, out_channels: int, nx: int,
+    ):
+        return MPGraphUNet(
+            self, in_channels=in_channels, out_channels=out_channels, nx=nx
+        )
+
+
+class MPNNGNN(nn.Module):
+    def __init__(
+        self,
+        node_in_channels,
+        node_hidden_channels,
+        edge_hidden_channels,
+        num_step_message_passing,
+        activation,
+        aggregator,
+        nx,
+    ):
+        """
+        MPNN is introduced in `Neural Message Passing for Quantum Chemistry
+        <https://arxiv.org/abs/1704.01212>`__.
+
+        Args:
+        ----------
+        node_in_channels : int
+            Size for the input node channels.
+        node_hidden_channels : int
+            Size for the hidden node representations.
+        edge_hidden_channels : int
+            Size for the hidden edge representations.
+        num_step_message_passing : int
+            Number of message passing steps.
+        activation: activation function
+        aggregator: aggregator type, one of 'sum', 'mean', 'max'
+        nx: number of horizontal grid points on each tile of the cubed sphere
+        """
+        super(MPNNGNN, self).__init__()
+        self.graph, self.edge_relation = build_dgl_graph_with_edge(nx_tile=nx)
+        self.project_node_features = nn.Sequential(
+            nn.Linear(node_in_channels, node_hidden_channels),
+            activation.instance,
+            nn.Linear(node_hidden_channels, node_hidden_channels),
+        )
+        self.num_step_message_passing = num_step_message_passing
+        edge_network = nn.Sequential(
+            nn.Linear(self.edge_relation.size(1), edge_hidden_channels),
+            activation.instance,
+            nn.Linear(
+                edge_hidden_channels, node_hidden_channels * node_hidden_channels
+            ),
+        )
+        self.gnn_layer = NNConv(
+            in_feats=node_hidden_channels,
+            out_feats=node_hidden_channels,
+            edge_func=edge_network,
+            aggregator_type=aggregator,
+        )
+        self.gru = nn.GRU(node_hidden_channels, node_hidden_channels)
+        self.relu = activation.instance
+        self.out_features = node_hidden_channels
+
+    def forward(self, in_node_features):
+        """Performs message passing and updates node representations.
+
+        Args:
+        g : DGL graph
+        in_node_features : Input node features.
+
+        Returns:
+        out_node_features : Output node representations.
+        """
```

Comment:
Please format this docstring to follow Google style.
Please add type hint to the call signature, including return type. I would normally expect something called `in_node_features` to be an integer, so specifying that it's a tensor would be helpful here.

Reply:
The arguments in the docstring don't match the arguments to the function.